### ISSUE TITLE
IndexOutOfBoundsException: setSpan (-1 ... 9) starts before 0

### ISSUE THREAD

**Issue by kbrosnan** [mozilla-mobile/fenix#16063]
https://sentry.prod.mozaws.net/operations/firefox-beta/issues/9776127/?referrer=github_plugin

```
IndexOutOfBoundsException: setSpan (-1 ... 9) starts before 0
    at android.text.SpannableStringInternal.checkRange(SpannableStringInternal.java:496)
    at android.text.SpannableStringInternal.setSpan(SpannableStringInternal.java:196)
    at android.text.SpannableStringInternal.setSpan(SpannableStringInternal.java:183)
    at android.text.SpannableString.setSpan(SpannableString.java:60)
    at androidx.core.app.AppOpsManagerCompat.addUnderline(AppOpsManagerCompat.java:4)
...
(16 additional frame(s) were not displayed)

setSpan (-1 ... 9) starts before 0
```

**Comment by pocmo** [comment]
It crashes with Japanese here (which definitely has incomplete HTML): `アカウントをお持ちでない方は、<u>アカウントを作成</u>して Firefox を端末間で同期しましょう。` - `HtmlCompat.fromHtml()` is probably broken since that should not crash? But the String may need fixing too.

**Comment by liuche** [comment]
What happened here is a combination of things: we did a string change on Nightly without renaming the string, the resulting behavior was a little undefined.

**Immediate fix** is one of the following:
- either uplift the correctly CDATA string to match the code
- OR back out the code change so that it uses the old substitution format

The easiest way to make sure this change is working is to delete all the localized versions of this string, and force them all to fall back onto the en-us string. This means the string will not be localized.

Afterwards, we'll want to investigate what side effects of not updating the string title (did this add the string to Pontoon? Is it partially translated across different languages?)

**Comment by liuche** [comment]
We should talk about what kinds of tests that we want to add for this kind of problem.

**Comment by liuche** [comment]
Steps to repro are also listed in #16016 and it is the same underlying cause.

The problem was that there was a mismatch between the localized strings and the code that processes the string, and the new strings were localized, but with the old code path.

We decided to go with uplifting the patch that parses the HTML string, and uses the new string in [en-US](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/res/values/strings.xml#L1180).

There are 3 categories that locales in 82 could fall under, due to this bug.
1) `sign_in_create_account_text` exists in locale and is HTML formatted (with `CDATA`) - this should show the correct string without crashing
2) `sign_in_create_account_text` does not exist at all (not localized yet) - this should show the (new) [en-US string](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/res/values/strings.xml#L1180)
3) `sign_in_create_account_text` has the [old string, localized](https://github.com/mozilla-mobile/fenix/blob/e514ede697075cc0eb1681a65dbf9a9f88398042/app/src/main/res/values/strings.xml#L1149) - this may crash. However, grepping the strings files in the app do not show any `sign_in_create_account_text` strings that don't start with `CDATA`

**Comment by ebalazs-sv** [comment]
Verified as fixed on Release 82.1.1 (Build #2015770923) GV 82.0 from 10/21 with Pixel 2 (Android 9), Nexus 9 (Android 7.1.1), Xiaomi Mi Pad 2 (Android 5.1), Google Pixel 4 XL (Android 11), OnePlus A3 (Android 6.0.1), Samsung Galaxy Tab S6 (Android 10).
I will remove the qa:needed label and close this issue.

**Comment by AlexanderChurbakov** [comment]
df
